php/class-cache.php

```diff
@@ -0,0 +1,1019 @@
+				'ids'   => array(
+					'type'        => 'array',
+					'default'     => array(),
+					'description' => __( 'The list of IDs to update.', 'cloudinary' ),
+				),
+				'state' => array(
+					'type'        => 'string',
+					'default'     => 'draft',
+					'description' => __( 'The state to update.', 'cloudinary' ),
+				),
+			),
+		);
+
+		return $endpoints;
+	}
+
+	/**
+	 * Get cached files for an cache point.
+	 *
+	 * @param \WP_REST_Request $request The request object.
+	 *
+	 * @return \WP_REST_Response
+	 */
+	public function rest_get_caches( $request ) {
+		$id           = $request->get_param( 'ID' );
+		$search       = $request->get_param( 'search' );
+		$page         = $request->get_param( 'page' );
+		$current_page = $page ? $page : 1;
+		$data         = $this->cache_point->get_cache_point_cache( $id, $search, $current_page );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Admin permission callback.
+	 *
+	 * Explicitly defined to allow easier testability.
+	 *
+	 * @return bool
+	 */
+	public function rest_can_manage_options() {
+		return current_user_can( 'manage_options' );
+	}
+
+	/**
+	 * Change the status of a cache_point.
+	 *
+	 * @param \WP_REST_Request $request Full details about the request.
+	 *
+	 * @return \WP_REST_Response|\WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function rest_disable_items( $request ) {
+		$ids   = $request['ids'];
+		$state = $request['state'];
+		foreach ( $ids as $id ) {
+			if ( 'delete' === $state ) {
+				wp_delete_post( $id );
+				continue;
+			}
+			$args = array(
+				'ID'          => $id,
+				'post_status' => $state,
+			);
+			wp_update_post( $args );
+		}
+
+		return $ids;
+	}
+
+	/**
+	 * Gets the upload method: url or file upload by determining if the site is accessible from the outside.
+	 *
+	 * @return string
+	 */
+	protected function get_upload_method() {
+		$method = get_transient( self::META_KEYS['upload_method'] );
+		if ( empty( $method ) ) {
+			$test_url = $this->media->base_url . '/image/fetch/' . $this->plugin->dir_url . 'no_file.svg';
+			$request  = wp_remote_head( $test_url );
+			$result   = wp_remote_retrieve_header( $request, 'x-cld-error' );
+			$method   = 'url';
+			if ( false !== strpos( $result, 'ERR_DNS_FAIL' ) ) {
+				$method = 'direct';
+			}
+			set_transient( self::META_KEYS['upload_method'], $method, DAY_IN_SECONDS );
+		}
+
+		return $method;
+	}
+
+	/**
+	 * Upload a static file.
+	 *
+	 * @param string $file The file path to upload.
+	 * @param string $url  The file URL to upload.
+	 *
+	 * @return string|\WP_Error
+	 */
+	public function sync_static( $file, $url ) {
+
+		$errored = get_transient( self::META_KEYS['upload_error'] );
+		if ( isset( $errored[ $file ] ) && 3 <= $errored[ $file ] ) {
+			unset( $errored[ $file ] );
+			set_transient( self::META_KEYS['upload_error'], $errored, 60 );
+
+			// Dont try again.
+			return new \WP_Error( 'upload_error' );
```

Comment:
@DavidCramer 

If we are using `WP_Error`, let's add more details about the error. Perhaps a descriptive message and reference for the errored file. What do you think?